src/extract/path/de.rs

```diff
@@ -0,0 +1,671 @@
+use crate::routing::UrlParams;
+use crate::util::ByteStr;
+use serde::{
+    de::{self, DeserializeSeed, EnumAccess, Error, MapAccess, SeqAccess, VariantAccess, Visitor},
+    forward_to_deserialize_any, Deserializer,
+};
+use std::fmt::{self, Display};
+
+/// This type represents errors that can occur when deserializing.
+#[derive(Debug, Eq, PartialEq)]
+pub(crate) struct PathDeserializerError(pub(crate) String);
+
+impl de::Error for PathDeserializerError {
+    #[inline]
+    fn custom<T: Display>(msg: T) -> Self {
+        PathDeserializerError(msg.to_string())
+    }
+}
+
+impl std::error::Error for PathDeserializerError {
+    #[inline]
+    fn description(&self) -> &str {
+        "path deserializer error"
+    }
+}
+
+impl fmt::Display for PathDeserializerError {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            PathDeserializerError(msg) => write!(f, "{}", msg),
+        }
+    }
+}
+
+macro_rules! unsupported_type {
+    ($trait_fn:ident, $name:literal) => {
+        fn $trait_fn<V>(self, _: V) -> Result<V::Value, Self::Error>
+        where
+            V: Visitor<'de>,
+        {
+            Err(PathDeserializerError::custom(concat!(
+                "unsupported type: ",
+                $name
+            )))
+        }
+    };
+}
+
+macro_rules! parse_single_value {
+    ($trait_fn:ident, $visit_fn:ident, $tp:literal) => {
+        fn $trait_fn<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+        where
+            V: Visitor<'de>,
+        {
+            if self.url_params.0.len() != 1 {
+                return Err(PathDeserializerError::custom(
+                    format!(
+                        "wrong number of parameters: {} expected 1",
+                        self.url_params.0.len()
+                    )
+                    .as_str(),
+                ));
+            }
+
+            let value = self.url_params.0[0].1.parse().map_err(|_| {
+                PathDeserializerError::custom(format!(
+                    "can not parse {:?} to a {}",
+                    self.url_params.0[0].1.as_str(),
+                    "bool"
```

Comment:
Seems this shouldn't always be `"bool"` 🤔 

```suggestion
                    $tp
```